website/tokens/__init__.py

```diff
@@ -0,0 +1,96 @@
+import httplib as http
+import functools
+import jwt
+from flask import request
+
+from framework.exceptions import HTTPError
+
+from website import settings
+from website.tokens import handlers
+from website.tokens.exceptions import TokenHandlerNotFound
+
+class TokenHandler(object):
+
+    HANDLERS = {
+        'approve_registration_approval': functools.partial(handlers.sanction_handler, 'registration', 'approve'),
+        'reject_registration_approval': functools.partial(handlers.sanction_handler, 'registration', 'reject'),
+        'approve_embargo': functools.partial(handlers.sanction_handler, 'embargo', 'approve'),
+        'reject_embargo': functools.partial(handlers.sanction_handler, 'embargo', 'reject'),
+        'approve_retraction': functools.partial(handlers.sanction_handler, 'retraction', 'approve'),
+        'reject_retraction': functools.partial(handlers.sanction_handler, 'retraction', 'reject')
+    }
+
+    def __init__(self, encoded_token=None, payload=None):
+
+        self.encoded_token = encoded_token
+        self.payload = payload
+
+    @staticmethod
```

Comment:
It seems unnecessary to make `encode` and `decode` static methods. The `TokenHandler`'s single responsibility is to route encoded actions to handler functions; the implementation for encoding and decoding is outside that responsibility. 

A practical consideration is that there may be other code in the future that may want to encode/decode JWTs without the need for any of the action/handler code.